src/datasource.ts

```diff
@@ -396,6 +404,31 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
     return [query, ...osFilters].filter(f => f).join(' AND ');
   }
 
+  addPPLAdHocFilters(queryString: string, adHocFilters: any) {
+    for (let i = 0; i < adHocFilters.length; i++) {
+      const { key, operator } = adHocFilters[i];
```

Comment:
Just curious: I notice in the existing `addPPLAdhocFilters` in `buildPPLQuery` that they're always using `adhocFilters[i].value`. How come we don't have to use `.value` here?

Reply:
using `let { value } = adHocFilters[i];` is shorter and easier to read in the later code since you can just use `value` instead of `adhocFilters[i].value`.